### PR TITLE
Use map of substitutions for replacing custom XML elements

### PR DESCRIPTION
**Pull request by jwakely**
This seems to work fine and generates identical lists, but let's leave it until after the Kona meeting before merging.

### CODE REVIEW

File: src/lists.cpp [hunk 1]
```diff
@@ -202,6 +202,21 @@ namespace {
 
 // ============================================================================================================
 
+namespace
+{
+   std::map<std::string_view, std::pair<std::string_view, std::string_view>, std::less<>> substitutions {
```
Comment (CaseyCarter) [inline src/lists.cpp:207]: Why not `unordered_map`?

File: src/lists.cpp [hunk 2]
```diff
@@ -202,6 +202,21 @@ namespace {
...
+      { "discussion", {"<p><b>Discussion:</b></p>", ""} },
+         { "resolution", {} },
+         { "rationale", {"<p><b>Rationale:</b></p>", ""} },
+         { "duplicate", {} },
+         { "note", { "<p><i>[", "]</i></p>\n"} },
+         { "superseded",
```
Comment (CaseyCarter) [inline src/lists.cpp:213]: Could we also add `<!ELEMENT superseded ANY>` to the DTD to keep it current?

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

File: src/lists.cpp (hunk 2 above)
Comment (jwakely) [reply src/lists.cpp:213]: Oh yes, I forgot about the DTD. Done, thanks!

File: src/lists.cpp (hunk 1 above)
Comment (jwakely) [reply src/lists.cpp:207]: :shrug: performance doesn't really matter here, and map was less typing! I can change it though.

Comment (CaseyCarter) [reply src/lists.cpp:207]: I don't care much, I just saw `map` and noticed we don't need ordering.

**Pull request merged**